ElementX/Sources/Services/Client/ClientProxy.swift

```diff
@@ -197,25 +197,40 @@ class ClientProxy: ClientProxyProtocol {
         return await waitForRoomSummary(with: result, name: expectedRoomName)
     }
     
-    func createRoom(with parameters: CreateRoomFlowParameters, userIDs: [String]) async -> Result<String, ClientProxyError> {
+    func createRoom(name: String, topic: String?, isRoomPrivate: Bool, userIDs: [String], roomImageMatrixUrl: String?) async -> Result<String, ClientProxyError> {
         let result: Result<String, ClientProxyError> = await Task.dispatch(on: clientQueue) {
             do {
-                let parameters = CreateRoomParameters(name: parameters.name,
-                                                      topic: parameters.topic,
-                                                      isEncrypted: parameters.isRoomPrivate,
+                let parameters = CreateRoomParameters(name: name,
+                                                      topic: topic,
+                                                      isEncrypted: isRoomPrivate,
                                                       isDirect: false,
-                                                      visibility: parameters.isRoomPrivate ? .private : .public,
-                                                      preset: parameters.isRoomPrivate ? .privateChat : .publicChat,
+                                                      visibility: isRoomPrivate ? .private : .public,
+                                                      preset: isRoomPrivate ? .privateChat : .publicChat,
                                                       invite: userIDs,
-                                                      avatar: nil)
+                                                      avatar: roomImageMatrixUrl)
                 let roomId = try self.client.createRoom(request: parameters)
                 return .success(roomId)
             } catch {
                 return .failure(.failedCreatingRoom)
             }
         }
         
-        return await waitForRoomSummary(with: result, name: parameters.name)
+        return await waitForRoomSummary(with: result, name: name)
+    }
+    
+    func uploadMedia(_ media: MediaInfo) async -> Result<String, ClientProxyError> {
+        guard let mimeType = media.mimeType else { return .failure(ClientProxyError.mediaFileError) }
+        return await Task.dispatch(on: clientQueue) {
+            do {
+                let data = try Data(contentsOf: media.mainURL)
+                let matrixUrl = try self.client.uploadMedia(mimeType: mimeType, data: [UInt8](data))
```

Comment:
```suggestion
                let matrixURL = try self.client.uploadMedia(mimeType: mimeType, data: [UInt8](data))
```
Sidenote, shouldn't this accept the original file's URL and let Rust do whatever it wants with the file rather than having us load the whole thing into memory on the app side? Not sure how related this binding is to your PR.

Reply:
I think you are right but It seems out of the scope of this PR, If you prefer I can create an issue about it so we won't lose track of it. Also probably the media upload Rust API will change anyways with the upload progress right?

Reply:
Yep yep, that would be great if you would 👍